common/theme_generator.py

```diff
@@ -70,10 +70,13 @@ def get_new_theme_path(self, name):
 
         full_path = os.path.join(sublime.packages_path(), path_in_packages)
 
-        with open(full_path, "wb") as out_f:
-            out_f.write(STYLES_HEADER.encode("utf-8"))
-            out_f.write(ElementTree.tostring(self.plist, encoding="utf-8"))
-
+        try:
+            with open(full_path, "wb") as out_f:
```

Comment:
Yeah, probably a good idea.

I not that good in python so possibly you you know good way to solve this.
If it was ruby we could do this

```python
# current
def get_new_theme_path(self, name):
    try:
        with open(full_path, "wb") as out_f:
            out_f.write(STYLES_HEADER.encode("utf-8"))
            out_f.write(ElementTree.tostring(self.plist, encoding="utf-8"))
    except PermissionError as e:
        sublime.ok_cancel_dialog("GitSavvy could not write file to path: \n{}".format(e))
        raise e
    except OSError as e:
        sublime.ok_cancel_dialog("GitSavvy encountered a disk error: \n{}".format(e))
        raise e

# What it could look like
def get_new_theme_path(self, name):
    common.file.open 
        with open(full_path, "wb") as out_f:
            out_f.write(STYLES_HEADER.encode("utf-8"))
            out_f.write(ElementTree.tostring(self.plist, encoding="utf-8"))

# in common.file
def open 
    try:
        yield
    except PermissionError as e:
        sublime.ok_cancel_dialog("GitSavvy could not write file to path: \n{}".format(e))
        raise e
    except OSError as e:
        sublime.ok_cancel_dialog("GitSavvy encountered a disk error: \n{}".format(e))
        raise e

```